src/main/java/org/broadinstitute/hellbender/tools/spark/sv/RunSGAViaProcessBuilderOnSpark.java

```diff
@@ -0,0 +1,522 @@
+     * Because some of the modules are designed to be run together in serial form in a unit (e.g. correction followed by indexing)
+     *   all return status and stdout stderr messages in this unit are logged.
+     * But only one file name is important for consumption by downstream units/modules, so only that file name is kept.
+     * The ctor and the add() method are designed asymmetrically, because when a unit is consisted of multiple modules,
+     *   only one generates the file name to be returned, and the other modules are simply to be logged.
+     */
+    @VisibleForTesting
+    static final class SGAUnitRuntimeInfoStruct implements Serializable{
+        private static final long serialVersionUID = 1L;
+
+        public final String fileNameToReturn;
+        public final List<Tuple2<String, Tuple3<ExternalCommandlineProgramModule.ReturnStatus, String, String>>> runtimeInfo;
+
+        public SGAUnitRuntimeInfoStruct(final String moduleName,
+                                        final String fileName,
+                                        final ExternalCommandlineProgramModule.ReturnStatus returnStatus,
+                                        final String stdoutMessage,
+                                        final String stderrMessage){
+            fileNameToReturn = fileName;
+            runtimeInfo = new ArrayList<>();
+            add(moduleName, returnStatus, stdoutMessage, stderrMessage);
+        }
+
+        public void add(final String moduleName, final ExternalCommandlineProgramModule.ReturnStatus returnStatus, final String stdoutMessage, final String stderrMessage){
+            runtimeInfo.add(new Tuple2<>(moduleName, new Tuple3<>(returnStatus, stdoutMessage, stderrMessage)));
+        }
+    }
+
+    /**
+     * Represents a collection of assembled contigs in the final output of "sga assemble".
+     */
+    @VisibleForTesting
+    static final class ContigsCollection implements Serializable{
+        private static final long serialVersionUID = 1L;
+
+        @VisibleForTesting
+        static final class ContigSequence implements Serializable{
+            private static final long serialVersionUID = 1L;
+
+            private final String sequence;
+            public ContigSequence(final String sequence){ this.sequence = sequence; }
+            public String getSequenceAsString(){ return sequence; }
+        }
+
+        @VisibleForTesting
+        static final class ContigID implements Serializable{
+            private static final long serialVersionUID = 1L;
+
+            private final String id;
+            public ContigID(final String idString) { this.id = idString; }
+            public String getId() { return id; }
+        }
+
+        private List<Tuple2<ContigID, ContigSequence>> contents;
+
+        public List<Tuple2<ContigID, ContigSequence>> getContents(){
+            return contents;
+        }
+
+        public ContigsCollection(final File fastaFile) throws IOException{
```

Comment:
To me it might be better to parse the file in the external class, and have the constructor here take a `List<String>`. That way this class is not tied at all to file I/O.